pkg/api/serve_test.go

```diff
@@ -57,10 +57,16 @@ type memCollector struct {
 	mu             sync.Mutex
 }
 
-func (m *memCollector) CollectEvent(ev stats.Event) {
+func (m *memCollector) CollectEvents(ev stats.Event, count uint64) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.Events = append(m.Events, &ev)
+	for i := 0; i < int(count); i++ {
+		m.Events = append(m.Events, &ev)
+	}
```

Comment:
This is the loop we had in the controller - I think it should be handled the same way - remove it in favor of reporting / update the event's count.